app/routes.js

```diff
@@ -25,8 +25,10 @@ import personUtils from './core/personutils';
  *
  * @return {boolean|null} returns true if hash mapping happened
  */
-export const requireAuth = (api) => (nextState, replace) => {
-  if (!api.user.isAuthenticated()) {
+export const requireAuth = (api, store) => (nextState, replace) => {
+  let { blip: state } = store.getState();
+
+  if (!state.isLoggedIn && !api.user.isAuthenticated()) {
```

Comment:
Sounds good. I'll remove it in this and any functions that have similar behaviour that you have commented on.